pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -102,12 +105,31 @@ func (al *ArtifactListing) ArtifactIDs(buildtimeClosure bool) ([]artifact.Artifa
 // artifact map by traversing the build plan from the terminal targets through
 // all of the runtime dependencies for each of the artifacts in the DAG.
 
+func newFilteredMapFromBuildPlan(build *model.Build, calculateBuildtimeClosure bool, cfg platformModel.Configurable) (artifact.Map, error) {
+	filtered, err := filterPlatformTerminal(build, cfg)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not filter terminals")
+	}
+
+	result, err := NewMapFromBuildPlan(build, calculateBuildtimeClosure, true, filtered)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not get map from build plan")
+	}
+
+	if _, ok := result[filtered.Tag]; !ok {
+		// This shouldn't happen, but better safe than sorry
+		return nil, errs.New("Resulting buildplan map is missing UUID: %s, map: %v", filtered.Tag, result)
+	}
+
+	return result[filtered.Tag], nil
+}
+
+type TerminalArtifactMap map[string]artifact.Map
+
 // Setting calculateBuildtimeClosure as true calculates the artifact map with the buildtime
 // dependencies. This is different from the runtime dependency calculation as it
 // includes ALL of the input artifacts of the step that generated each artifact.
-func newMapFromBuildPlan(build *model.Build, calculateBuildtimeClosure bool, cfg platformModel.Configurable) (artifact.Map, error) {
-	res := make(artifact.Map)
-
+func NewMapFromBuildPlan(build *model.Build, calculateBuildtimeClosure bool, filterStateToolArtifacts bool, filterTerminal *model.NamedTarget) (TerminalArtifactMap, error) {
```

Comment:
nit: We should update this docstring and it should start with the function name.

I know you didn't write the docstring. I did so that's on me, but we should still improve it here if we can.